frontend/src/components/samples/SamplesListContent.js

```diff
@@ -196,17 +212,43 @@ const SamplesListContent = ({
     (mergedListQueryParams(SAMPLE_FILTERS, filters, sortBy), template)
       .then(response => response)
 
-  const columns = getTableColumns(containersByID, individualsByID, projectsByID, sampleKinds)
-  .map(c => Object.assign(c, getFilterProps(
+  const nFilters = getNFilters(filters)
+
+  // Listen to the changes in toggle to get new columns and set filters accordingly
+  useEffect(() => {
+    // Get the new columns depending on the new option
+    setColumns(getTableColumns(containersByID, individualsByID, projectsByID, sampleKinds, toggleOption))
+
+    // Only apply filters if the new selected option is different from the current one
+    if (getCurrentToggleOption() !== toggleOption){
+      // Need to reset the sortBy when there's a new toggle option to avoid problems
+      setSortByWithoutList('', '')
```

Comment:
Maybe you should call this clearSortBy. The List would otherwise be required ... From Antd, null are used instead of empty string ... I would prefer having null instead of empty strings if it works. If  as it seem this is a clearSortBy you can remove the params (i don't think we will ever use this for something else)